apps/emqx_rule_engine/src/emqx_rule_runtime.erl

```diff
@@ -89,15 +90,21 @@ do_apply_rule(#rule{id = RuleId,
                     on_action_failed = OnFailed,
                     actions = Actions}, Input) ->
     {Selected, Collection} = ?RAISE(select_and_collect(Fields, Input),
+                                        emqx_rule_metrics:inc_rules_exception(RuleId),
                                         {select_and_collect_error, {_EXCLASS_,_EXCPTION_,_ST_}}),
     ColumnsAndSelected = maps:merge(Input, Selected),
     case ?RAISE(match_conditions(Conditions, ColumnsAndSelected),
+                emqx_rule_metrics:inc_rules_exception(RuleId),
```

Comment:
Could we move these `inc_rules_exception` lines into the `catch` clause of `try apply_rule_discard_result(Rule, Input)`?